arelle/plugin/validate/DBA/rules/fr.py

```diff
@@ -512,6 +512,44 @@ def rule_fr56(
             )
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+)
+def rule_fr58(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    DBA.FR58: If fsa:ReportingResponsibilitiesAccordingToTheDanishExecutiveOrderOnApprovedAuditorsReportAudit is tagged then one of the following concepts MUST also be tagged:
+    fsa:ReportingResponsibilitiesAccordingToTheDanishExecutiveOrderOnApprovedAuditorsReportsEspeciallyTheCriminalCodeAndFiscalTaxAndSubsidyLegislationAudit
+    fsa:ReportingResponsibilitiesAccordingToTheDanishExecutiveOrderOnApprovedAuditorsReportsEspeciallyTheCompaniesActOrEquivalentLegislationThatTheCompanyIsSubjectToAudit
+    fsa:ReportingResponsibilitiesAccordingToTheDanishExecutiveOrderOnApprovedAuditorsReportsEspeciallyLegislationOnFinancialReportingIncludingAccountingAndStorageOfAccountingRecordsAudit
+    fsa:ReportingResponsibilitiesAccordingToTheDanishExecutiveOrderOnApprovedAuditorsReportsEspeciallyOtherMattersAudit
+
+    """
+    modelXbrl = val.modelXbrl
+    indicatorFacts = modelXbrl.factsByQname.get(pluginData.reportingResponsibilitiesOnApprovedAuditorsReportAuditQn)
+    if indicatorFacts is not None:
+        foundFact = False
+        for qname in pluginData.declarationObligationQns:
+            facts = modelXbrl.factsByQname.get(qname)
+            if facts is not None and foundFact is False:
```

Comment:
Why is `foundFact is False` needed?

Reply:
I was just attempting to shortcut any further processing if a suitable fact had already been found.